src/app/modules/payments/stripe.controller.ts

```diff
@@ -227,66 +237,88 @@ export const handleStripeEventUpdates = [
   _handleStripeEventUpdates,
 ]
 
-// TODO: Refactor for use in static payment url implementation
 export const checkPaymentReceiptStatus: ControllerHandler<{
   formId: string
-  submissionId: string
+  paymentId: string
 }> = async (req, res) => {
-  const { formId, submissionId } = req.params
+  const { formId, paymentId } = req.params
   logger.info({
     message: 'getPaymentStatus endpoint called',
     meta: {
       action: 'getPaymentStatus',
-      formId: formId,
-      submissionId: submissionId,
+      formId,
+      paymentId,
     },
   })
 
-  return PaymentService.findPaymentBySubmissionId(submissionId)
+  return PaymentService.findPaymentByPaymentIntentId(paymentId)
     .map((payment) => {
       logger.info({
-        message: 'Received payment object with receipt url from Stripe webhook',
+        message: 'Found paymentId in payment document',
         meta: {
           action: 'checkPaymentReceiptStatus',
           payment,
         },
       })
 
+      if (!payment.completedPayment?.receiptUrl) {
+        return res.status(StatusCodes.NOT_FOUND).json({ isReady: false })
+      }
       return res.status(StatusCodes.OK).json({ isReady: true })
+
+      // no payment found, perhaps webhook from stripe has not arrived
+      // trigger a manual sync flow
+      // TODO: manual sync flow not available after https://github.com/opengovsg/FormSG/pull/5937
+      // return StripeService.getPaymentFromLatestSuccessfulCharge(
+      //   formId,
+      //   paymentId,
+      // )
+      //   .map((payment) => {
+      //     // has confirmedPayment but no receiptUrl, system is desync-ed
+      //     if (!payment.completedPayment?.receiptUrl) {
+      //       logger.error({
+      //         message: 'has confirmedPayment but no receiptUrl',
+      //         meta: {
+      //           action: 'checkPaymentReceiptStatus',
+      //           payment,
+      //           paymentId,
+      //           formId,
+      //         },
+      //       })
+      //       return res
+      //         .status(StatusCodes.INTERNAL_SERVER_ERROR)
+      //         .json({ message: 'Missing receipt url' })
+      //     }
+      //     return res.status(StatusCodes.OK).json({ isReady: true })
+      //   })
+      //   .mapErr((error) => {
+      //     return res.status(StatusCodes.NOT_FOUND).json({ message: error })
+      //   })
```

Comment:
discussion point: do we need to have a manual sync flow? or should we be polling and ask the user to refresh / click a button to refetch? i feel like the latter is better haha

Reply:
Yeah there's a polling happening (provided by react-query).
Some other details, the BE is returning `404` which react-query will recognize as a failure and retry until it succeeds.

I'm also on the bus for polling initially, but iirc @timotheeg mentioned that we should trigger a sync.

Reply:
hmm, ok, we can track this in a separate issue?

Reply:
Opened here opengovsg/formsg-private/issues/143